src/frontend/apps/lti_site/apps/deposit/components/Dashboard/common/DepositedFileRow/index.tsx

```diff
@@ -58,131 +59,68 @@ export const DepositedFileRow = ({ file }: DepositedFileProps) => {
-              >
-                {size === 'medium'
-                  ? truncateFilename(file.filename, 40)
-                  : size === 'large'
-                  ? truncateFilename(file.filename, 60)
-                  : file.filename}
-              </Text>
-            </Grid>
+              {uploadedOnDate}&nbsp;{uploadedOnTime}
+            </Text>
           </Box>
-          {file.upload_state === 'ready' ? (
-            <Box align="center" justify="end" direction="row" gap="small">
-              <Box
-                align="center"
-                justify="center"
-                background="blue-active"
-                pad={{ horizontal: 'medium', vertical: 'xsmall' }}
-                round="xsmall"
-              >
-                <Anchor
-                  onClick={markFileAsRead}
-                  download
-                  color="white"
-                  a11yTitle={intl.formatMessage(messages.labelDownload)}
-                  label={intl.formatMessage(messages.labelDownload)}
-                  style={{ fontFamily: 'Roboto-Medium' }}
-                  title={intl.formatMessage(messages.labelDownload)}
-                  href={file.url}
-                />
-              </Box>
-            </Box>
-          ) : (
-            <Box justify="end" direction="row" gap="small">
+        )}
+        <Box align="end" justify="end" gap="small" flex={true}>
+          <Box
+            justify="end"
+            background={
+              file.upload_state === 'ready' ? 'blue-active' : undefined
+            }
+            pad={{
+              horizontal: file.upload_state === 'ready' ? 'medium' : undefined,
+              vertical: 'xsmall',
+            }}
+            round="xsmall"
+          >
+            {file.upload_state === 'ready' ? (
+              <Anchor
+                onClick={markFileAsRead}
+                download
+                color="white"
+                a11yTitle={intl.formatMessage(messages.labelDownload)}
+                label={intl.formatMessage(messages.labelDownload)}
+                style={{ fontFamily: 'Roboto-Medium' }}
+                title={intl.formatMessage(messages.labelDownload)}
+                href={file.url}
+              />
+            ) : (
               <Text>{file.upload_state}</Text>
-            </Box>
-          )}
-        </Grid>
-      )}
+            )}
+          </Box>
+        </Box>
+      </Box>
+
+      <Box direction="row" fill>
+        <Box justify="start" flex={true}>
+          <Text title={file.filename} weight={file.read ? 'normal' : 'bolder'}>
+            {size === 'medium'
+              ? truncateFilename(file.filename, 40)
+              : size === 'large'
+              ? truncateFilename(file.filename, 60)
+              : file.filename}
+          </Text>
+        </Box>
+        <Box align="end" justify="end" flex={true}>
+          <Text size="small">{bytesToSize(file.size)}</Text>
+        </Box>
+      </Box>
```

Comment:
Are you sure you want to set flex on both Box ? This will spilt the parent container in 2 in the middle.
If one context might be bigger than the other one it might be better to skrink one and flex the other no ?

Reply:
I suppose, yes…
I shrinked the smallest one, which will always have the same width.